plugins/github/current_github.ml

```diff
@@ -17,7 +17,12 @@ module Metrics = struct
     Counter.v_label ~label_name:"event" ~help ~namespace ~subsystem "webhook_events_total"
 end
 
-let webhook = object
+let validate_webhook_payload webhook_secret body headers =
+  let request_signature = Option.value ~default:"<empty>" (Cohttp.Header.get headers "X-Hub-Signature-256") in
+  let signature = "sha256=" ^ Hex.show @@ Hex.of_cstruct @@ Mirage_crypto.Hash.SHA256.(hmac ~key:(Cstruct.of_string webhook_secret) (Cstruct.of_string body)) in
+  String.equal signature request_signature
```

Comment:
Does this need to use `Eqaf.equal` or some other constant-time string comparison function instead?

Reply:
Indeed, the GitHub docs state:

> Using a plain == operator is not advised. A method like secure_compare performs a "constant time" string comparison, which helps mitigate certain timing attacks against regular equality operators.

https://docs.github.com/en/developers/webhooks-and-events/webhooks/securing-your-webhooks

Reply:
Thanks I didn't know about `Eqaf.equal` and that whole package.